CMakeLists.txt

```diff
@@ -59,7 +59,7 @@ target_include_directories(lobster PUBLIC lobster/include lobster/src lobster/ex
 add_library(lobster-impl STATIC src/lobster_impl.cpp)
 target_link_libraries(lobster-impl PRIVATE lobster)
 
-set(CMAKE_CXX_CLANG_TIDY clang-tidy -checks=*)
+set(CMAKE_CXX_CLANG_TIDY clang-tidy -checks=cppcoreguidelines-*,clang-analyzer-*,readability-*,performance-*,portability-*,concurrency-*,modernize-*)
```

Comment:
How many issues does this currently show? We want it to catch potential problems, not mindless substitution of newer types (such as the std::array change).. I believe `cppcoreguidelines` and/or `modernize` are likely going to complain about the use of raw pointers in a million places and we don't want to fix those.

Reply:
Actually it is not that much
```
/dvl/treesheets/src/main.cpp:10:12: warning: variable 'g_grid_margin' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
static int g_grid_margin = 1;
           ^
/dvl/treesheets/src/main.cpp:11:12: warning: variable 'g_cell_margin' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
static int g_cell_margin = 2;
           ^
/dvl/treesheets/src/main.cpp:12:12: warning: variable 'g_margin_extra' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
static int g_margin_extra = 2;  // TODO, could make this configurable: 0/2/4/6
           ^
/dvl/treesheets/src/main.cpp:13:12: warning: variable 'g_line_width' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
static int g_line_width = 1;
           ^
/dvl/treesheets/src/main.cpp:14:12: warning: variable 'g_selmargin' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
static int g_selmargin = 2;
           ^
/dvl/treesheets/src/main.cpp:15:12: warning: variable 'g_deftextsize' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
static int g_deftextsize = 12;
           ^
/dvl/treesheets/src/main.cpp:15:28: warning: 12 is a magic number; consider replacing it with a named constant [cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers]
static int g_deftextsize = 12;
                           ^
/dvl/treesheets/src/main.cpp:16:12: warning: use a trailing return type for this function [modernize-use-trailing-return-type]
static int g_mintextsize() { return g_deftextsize - 8; }
       ~~~ ^
       auto                -> int
/dvl/treesheets/src/main.cpp:16:53: warning: 8 is a magic number; consider replacing it with a named constant [cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers]
static int g_mintextsize() { return g_deftextsize - 8; }
                                                    ^
/dvl/treesheets/src/main.cpp:17:12: warning: use a trailing return type for this function [modernize-use-trailing-return-type]
static int g_maxtextsize() { return g_deftextsize + 32; }
       ~~~ ^
       auto                -> int
/dvl/treesheets/src/main.cpp:17:53: warning: 32 is a magic number; consider replacing it with a named constant [cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers]
static int g_maxtextsize() { return g_deftextsize + 32; }
                                                    ^
/dvl/treesheets/src/main.cpp:20:12: warning: variable 'g_scrollratecursor' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
static int g_scrollratecursor = 240;  // FIXME: must be configurable
           ^
/dvl/treesheets/src/main.cpp:20:33: warning: 240 is a magic number; consider replacing it with a named constant [cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers]
static int g_scrollratecursor = 240;  // FIXME: must be configurable
                                ^
/dvl/treesheets/src/main.cpp:21:12: warning: variable 'g_scrollratewheel' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
static int g_scrollratewheel = 2;     // relative to 1 step on a fixed wheel usually being 120
           ^
/dvl/treesheets/src/main.cpp:273:12: warning: destructor of 'TreeSheetsScriptImpl' is public and non-virtual [cppcoreguidelines-virtual-class-destructor]
    struct TreeSheetsScriptImpl;
           ^
/dvl/treesheets/src/main.cpp:273:12: note: make it public and virtual
/dvl/treesheets/src/main.cpp:289:20: warning: variable 'sys' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
    static System *sys;
                   ^
/dvl/treesheets/src/main.cpp:289:20: warning: variable 'sys' provides global access to a non-const object; consider making the pointed-to data 'const' [cppcoreguidelines-avoid-non-const-global-variables]
/dvl/treesheets/src/main.cpp:308:33: warning: variable 'sys' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
treesheets::System *treesheets::sys = nullptr;
                                ^
/dvl/treesheets/src/main.cpp:308:33: warning: variable 'sys' provides global access to a non-const object; consider making the pointed-to data 'const' [cppcoreguidelines-avoid-non-const-global-variables]
/dvl/treesheets/src/main.cpp:309:46: warning: variable 'tssi' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables]
treesheets::TreeSheetsScriptImpl treesheets::tssi;
                                             ^
/dvl/treesheets/src/myapp.h:177:9: warning: deleting a pointer through a type that is not marked 'gsl::owner<>'; consider using a smart pointer instead [cppcoreguidelines-owning-memory]
        DELETEP(sys);
        ^
/dvl/treesheets/src/tools.h:31:13: note: expanded from macro 'DELETEP'
            delete p; \
            ^      ~
/dvl/treesheets/src/main.cpp:289:5: note: variable declared here
    static System *sys;
    ^~~~~~~~~~~~~~~~~~
```

Reply:
most of those `g_` variables could be moved into `System`, but `sys` itself we want to keep global I think.

`use a trailing return type for this function` seems not worth it, should be disabled.